### PR TITLE
restaurant customers

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cses-1619.mdx [hunk 1]
```diff
@@ -143,115 +143,106 @@ If we put the start and end points into a list and sort them, all we need to do
...
+		int[][] times = new int[2 * n][];
+		for (int i = 0; i < n; i++) {
+			StringTokenizer cus = new StringTokenizer(read.readLine());
+			times[2 * i] = new int[] {Integer.parseInt(cus.nextToken()), 1};
+			times[2 * i + 1] = new int[] {Integer.parseInt(cus.nextToken()), -1};
+		}
+ 
+		Arrays.sort(
+			times,
+			(t1, t2) -> t1[0] != t2[0] ? t1[0] - t2[0] : t1[1] - t2[1]
```
Comment (envyaims) [inline solutions/cses-1619.mdx:199]: maybe add a comment here explaining what it does since its kind of compact

File: solutions/cses-1619.mdx [hunk 2]
```diff
@@ -260,53 +251,52 @@ Coordinate compress interval endpoints and only compute the prefix sum array for
...
-	vector<long long> pfx(2*N + 1); // prefix sum array
-	vector<int> arr(2*N);
-
-	for (int i = 0; i < 2*N; ++i) { // coordinate compression
-		if(i == 0) { customers[i].first = 0; }
-		else if(customers[i].first > curr) { l++; curr = customers[i].first; } // we move a pointer
-		arr[l+1] += customers[i].second;
-		// pfx[l+1] += customers[i].second; implementation without arr, merges all interval end and starts here
+	int at = 0;
+	vector<int> ppl_change(2 * n);
```
Comment (envyaims) [inline solutions/cses-1619.mdx:275]: wat this store

File: solutions/cses-1619.mdx [hunk 3]
```diff
@@ -260,53 +251,52 @@ Coordinate compress interval endpoints and only compute the prefix sum array for
...
+			at++;
+			curr = times[i].first;
+		}
+		ppl_change[at + 1] += times[i].second;
 	}
 
-	for (int i = 1; i < 2*N + 1; ++i) {
-		pfx[i] = arr[i] + pfx[i-1];
-		// pfx[i] += pfx[i-1]; implementation without arr
+	vector<int> ppl_amt(2 * n + 1);
```
Comment (envyaims) [inline solutions/cses-1619.mdx:287]: make it explicit this is the prefix sum array

**Review by envyaims — CHANGES_REQUESTED** [review]
REDACTED

File: solutions/cses-1619.mdx [hunk 4]
```diff
@@ -143,115 +143,106 @@ If we put the start and end points into a list and sort them, all we need to do
...
 
 </CPPSection>
-	
 <JavaSection>
-	
-<Warning>
-Java solutions using this implementation will TLE on test case #6.
-</Warning>
+
+If we use the same implementation as the C++ version, CSES will give a TLE on test case 6.
```
Comment (ryanchou-dev) [inline solutions/cses-1619.mdx:179]:
```suggestion
If we use the same implementation as the C++ version, CSES will give a TLE verdict on test case 6.
```

**Review by ryanchou-dev — APPROVED** [review]
address envy's comments as well!!

File: solutions/cses-1619.mdx [hunk 5]
```diff
@@ -260,53 +249,55 @@ Coordinate compress interval endpoints and only compute the prefix sum array for
...
-	vector<long long> pfx(2*N + 1); // prefix sum array
-	vector<int> arr(2*N);
-
-	for (int i = 0; i < 2*N; ++i) { // coordinate compression
-		if(i == 0) { customers[i].first = 0; }
-		else if(customers[i].first > curr) { l++; curr = customers[i].first; } // we move a pointer
-		arr[l+1] += customers[i].second;
-		// pfx[l+1] += customers[i].second; implementation without arr, merges all interval end and starts here
+	int at = 0;
+	// Stores how much the # of people in the restaurant changes each time step
```
Comment (envyaims) [inline solutions/cses-1619.mdx:273]: still kinda weird wording

**Review by envyaims — APPROVED** [review]
ok 😛

File: solutions/cses-1619.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/cses-1619.mdx:273]: `compress` is definitely not a better name

Comment (envyaims) [reply solutions/cses-1619.mdx:273]: its more straight forward

Comment (envyaims) [reply solutions/cses-1619.mdx:273]: is it the difference in the numbers of customers?

Comment (SansPapyrus683) [reply solutions/cses-1619.mdx:273]: i guess?

Comment (envyaims) [reply solutions/cses-1619.mdx:273]: better off making the comment that then lol

Comment (SansPapyrus683) [reply solutions/cses-1619.mdx:273]: it's the same thing as the comment

Comment (envyaims) [reply solutions/cses-1619.mdx:273]: difference across the numbers of customers?? i think its better wording than the current version..

Comment (SansPapyrus683) [reply solutions/cses-1619.mdx:273]: at this point i feel like it's your word against mine
@ryanchou-dev

**Pull request merged**